Core/NoOpRunner.Core/NoOpRunner.cs

```diff
@@ -193,57 +208,47 @@ public void HandleKeyPress(KeyPress keyPress)
-            GamePlatforms.AddShape(aShape1);
-            GamePlatforms.AddShape(aShape2);
-            GamePlatforms.AddShape(aShape3);*/
-
-            AbstractFactory impassableFactory = FactoryProducer.GetFactory(passable: false);
-            BaseShape firstPlatform = impassableFactory.CreateStaticShape(Shape.Platform,
-                new CombinedGenerationStrategy(), 0, 0, GameSettings.HorizontalCellCount,
-                GameSettings.VerticalCellCount / 3);
-            PlatformsContainer.AddShape(firstPlatform);
-
-            AbstractFactory passableFactory = FactoryProducer.GetFactory(passable: true);
-            BaseShape secondPlatform = passableFactory.CreateStaticShape(Shape.Platform,
-                new PlatformerGenerationStrategy(), 0, GameSettings.VerticalCellCount / 3 + 1,
-                GameSettings.HorizontalCellCount, GameSettings.VerticalCellCount * 2 / 3);
-            PlatformsContainer.AddShape(secondPlatform);
-
-            // Generate the player above the first platform
-            Player = new Player(secondPlatform.CenterPosX, secondPlatform.CenterPosY + 1);
-
-            BaseShape thirdPlatform = passableFactory.CreateStaticShape(Shape.Platform,
-                new RandomlySegmentedGenerationStrategy(), 0, GameSettings.VerticalCellCount * 2 / 3 + 1,
-                GameSettings.HorizontalCellCount, GameSettings.VerticalCellCount - 3);
-            PlatformsContainer.AddShape(thirdPlatform);
-
-            var coordinates = firstPlatform.GetCoords();
-            int[] xCoords = coordinates.Item1;
-            int[] yCoords = coordinates.Item2;
-            int randomLocation = RandLocation(xCoords, yCoords);
-            PowerUp testPowerUp =
-                new PowerUp(xCoords[randomLocation], yCoords[randomLocation] + 2, PowerUps.Double_Jump);
-            PowerUpsContainer.AddShape(testPowerUp);
+            var gameeStateBuilder = new GameStateBuilder();
+            var initialGameState = gameeStateBuilder.Configure()
+                .InitializeMap(GameSettings.HorizontalCellCount, GameSettings.VerticalCellCount)
+                .InitializePowerUps(GameSettings.HorizontalCellCount, GameSettings.VerticalCellCount)
+                .AddImpassableShape(f => f.CreateStaticShape(Shape.Platform, new CombinedGenerationStrategy(), 0, 0, GameSettings.HorizontalCellCount, GameSettings.VerticalCellCount / 3))
+                .AddPassableShape(f => f.CreateStaticShape(Shape.Platform, new PlatformerGenerationStrategy(), 0, GameSettings.VerticalCellCount / 3 + 1, GameSettings.HorizontalCellCount, GameSettings.VerticalCellCount * 2 / 3))
+                .AddPassableShape(f => f.CreateStaticShape(Shape.Platform, new RandomlySegmentedGenerationStrategy(), 0, GameSettings.VerticalCellCount * 2 / 3 + 1, GameSettings.HorizontalCellCount, GameSettings.VerticalCellCount - 3))
+                .AddPlayer(platforms => platforms.First(p => p.GetType() == typeof(ImpassablePlatform)))
```

Comment:
The player should be generated above the second (middle) passable platform, not the first one